addon/mixins/dom.js

```diff
@@ -141,14 +141,25 @@ export default Mixin.create({
 
   willDestroyElement() {
     this._super(...arguments);
+    this._removeEventListeners();
+  },
+
+  willDestroy() {
+    this._super(...arguments);
+    this._removeEventListeners();
+  },
 
+  /**
+   @private
+   */
+  _removeEventListeners() {
     if (this._listeners) {
```

Comment:
check !== undefined or Array.isArray(), don't use falsy, this is internal we control how it is initialized checked.